server/store/kvstore/legal_hold.go

```diff
@@ -86,8 +82,8 @@ func (kvs Impl) GetLegalHoldByID(id string) (*model.LegalHold, error) {
 }
 
 func (kvs Impl) UpdateLegalHold(lh, oldValue model.LegalHold) (*model.LegalHold, error) {
-	if err := lh.IsValidForCreate(); err != nil {
-		return nil, errors.Wrap(err, "LegalHold is not valid for create")
+	if !mattermostModel.IsValidId(lh.ID) {
+		return nil, fmt.Errorf("legal hold does not have a valid id")
```

Comment:
This is already taken care of in `func (ulh UpdateLegalHold) IsValid` that is called from the API level. 

Why are we rechecking the ID?

Reply:
You're right that this seems redundant.